Sources_v3/Controllers/ChannelController/ChannelController.swift

```diff
@@ -114,9 +114,27 @@ public typealias ChatChannelController = _ChatChannelController<DefaultExtraData
 public class _ChatChannelController<ExtraData: ExtraDataTypes>: DataController, DelegateCallable, DataStoreProvider {
     /// The ChannelQuery this controller observes.
     @Atomic public private(set) var channelQuery: ChannelQuery<ExtraData>
+    
+    /// Flag indicating whether channel is created on backend. We need this flag to restrict channel modification requests
+    /// before channel is created on backend.
+    /// There are 2 ways of creating new channel:
+    /// 1. Direct message channel.
+    /// In this case before channel creation `cid` on `channelQuery` will be nil cause it will be generated on backend.
+    /// 2. Channels with client generated `id`.
+    /// In this case `cid` on `channelQuery `will be valid but all channel modifications will
+    /// fail because channel with provided `id` will be missing on backend side.
+    /// That is why we need to check both flag and valid `cid` before modifications.
+    private var isChannelAlreadyCreated: Bool
 
     /// The identifier of a channel this controller observes.
-    private var channelId: ChannelId { channelQuery.cid }
+    /// To perform any modifications on channel it needs to be created on backend side and have a valid cid.
+    /// If channel is not yet created on backend it will be `nil` here even though it's available in `ChannelQuery`.
+    /// Using `cid` value from here will guarantee that channel is created on backend and it's safe to create another
+    /// controllers where `cid` is required with this value.
+    public var cid: ChannelId? {
+        guard isChannelAlreadyCreated else { return nil }
+        return channelQuery.cid
+    }
```

Comment:
I'm not sure about this assumption. The fact, that the channel hasn't been created yet, doesn't mean, its `cid` is invalid.

Let's summarize this:
- if you know the channel exists, you use `channelController(for cid: ChannelId)`, `controller.cid` will always return a valid cid
- for 1-1 channels, you use `channelController(createDirectMessageChannelWith members: Set<UserId>)`, before the channel is created, `controller.cid` will return `nil`.
- when you create a channel with `channelController(createChannelWithId cid: ChannelId)`, I don't see a reason why `controller.cid` should return `nil`, because the `cid` is known at the time. If you use the `cid` before the channel is created (you will need to check for `controller.state == .remoteDataFetched`), it's a programmer error 🤷‍♂️ 

There are two things mixed up together here. Exposing an invalid `cid` and a channel that doesn't exist yet.  The first version of the PR returned an invalid cid for some cases and the user had no chance to learn about it.  Having `cid` optional fixes this problem. IMO it shouldn't be related to the problem of mutating non-existing channel. It's already handled in the action methods.

Reply:
But if we will look into case similar to the one you provided previously:
```
let controller1 = client.channelController(createChannelWithId: cid)
let controller2 = client.channelController(for: controller1.cid)
```
It will be possible to perform modifications on `controller2`.
I can agree that it's programmers error, but we are helping to avoid that.
And I see it logically correct: we have cid in `ChannelQuery` cause it's valid but `ChannelController` doesn't have one yet cause it's not observing any channel atm.

Reply:
The case I originally pointed out with the following:
```swift
let controller1 = client.channelController(createDirectMessageChannelWith: members)
let controller2 = client.channelController(for: controller1.cid)
```
The main difference is that in this case, `controller1` _looks like_ it has a valid `cid` but it doesn't.

The case you present is basically this:
```swift
let newCid = <some new cid>
let controller1 = client.channelController(createChannelWithId: newCid)
let controller2 = client.channelController(for: newCid)
```
If someone does this, then it's a clear misusage of the SDK.